src/project_generator.rs

```diff
@@ -36,6 +39,18 @@ impl Generator {
             Some(d) => d,
             None => project_title.clone(),
         };
+        
+        if fs::metadata(&directory).map(|metadata| metadata.is_dir()).unwrap_or(false) {
+            println!(
+                "{}",
+                format!(
+                    "{} Directory '{}' already exists. Please use other name for directory.",
+                    "🚫 Error:".red(),
+                    &directory.bold(),
+                )
+            );
+            process::exit(1);
+        }
```

Comment:
Can you make the function `Generator::new()` to return a Result and a new custom Error (see `errors.rs`) so that we can use the logger for consistency and not use `process::exit(1)`, but rather match on error and shut down gracefully?